spring-cloud-azure-samples/spring-cloud-azure-cosmosdb-mongodb-sample/README.adoc

```diff
@@ -0,0 +1,62 @@
+= Spring Cloud Azure CosmosDB using MongoDB API Sample
+
+This starter is based on `spring-boot-starter-data-mongodb`. It fetches and auto configures data source properties via Azure CosmosDB Management SDK.
+
+Maven coordinates:
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.microsoft.azure</groupId>
+    <artifactId>spring-cloud-azure-starter-cosmosdb-mongodb</artifactId>
+</dependency>
+----
+
+Gradle coordinates:
+
+[source]
+----
+dependencies {
+    compile group: 'com.microsoft.azure', name: 'spring-cloud-azure-starter-cosmosdb-mongodb'
+}
+----
+
+== How to run
+
+1.  Create azure credential file. Please see https://github.com/Azure/azure-libraries-for-java/blob/master/AUTH.md[how to create credential file]
++
+....
+$  az account set --subscription <name or id>
+$  az ad sp create-for-rbac --sdk-auth > my.azureauth
+....
++
+Make sure `my.azureauth` is encoded with UTF-8.
+
+2.  Put auth file under `src/main/resources/`.
+3.  Update link:src/main/resources/application.properties[application.properties]
++
+....
+spring.cloud.azure.credentialFilePath=[credential-file]
+spring.cloud.azure.resourceGroup=[resource-group]
+spring.cloud.azure.region=[region]
+
+# Auto create cosmosdb account and database if not existed
+spring.cloud.azure.cosmosdb.account-name=[database-account-name]
+spring.cloud.azure.cosmosdb.database=[database-name]
+spring.cloud.azure.cosmosdb.read-replication=[read-replication-region]
```

Comment:
Same as above for `read-replication`

Reply:
Does this region have to a different region?

Reply:
No, they don't have to.